File: js/src/movieService.js
var request = require('request');

function MovieService () {
  var serviceBaseUrl = 'http://hickory-snap.hyperdev.space/';

  this.getAllMovies = function (callback) {
    request.get(
      serviceBaseUrl + 'all',
      function (err, res, json) {
        if (err) {
          callback(err)
        } else if (res.statusCode == 200) {
          callback(null, json);
        } else {
          callback("Unexpected status code: " + res.statusCode);
        }
      }
    );
  };
  this.getCurrentRentals = function (username, callback) {
    request.get(
      serviceBaseUrl + 'rentals?username=' + username,
      function (err, res, json) {
        if (err) {
          callback(err)
        } else if (res.statusCode == 200) {
          callback(null, json);
        } else {
          callback("Unexpected status code: " + res.statusCode);
        }
      }
    );
  };

  this.addMovie = function (username, movie, callback) {
    request.post(
      serviceBaseUrl + 'addMovie',
      {json: { username: username, movie: movie } },
      function (err, res, json) {
        if (err) {
          callback(err)
        } else if (res.statusCode == 200) {
          callback(null, json);
        } else {
          callback("Unexpected status code: " + res.statusCode);
        }
      }
    );
  };
  this.rentMovie = function (username, movieId, callback) {
    request.post(
      serviceBaseUrl + 'rentMovie',
      { json: { username: username, id: movieId } },
      function (err, res, json) {
        if (err) {
          callback(err)
        } else if (res.statusCode == 200) {
          callback(null, json);
        } else {
          callback("Unexpected status code: " + res.statusCode);
        }
      }
    );
  };
}
module.exports = new MovieService();

File: java/src/main/java/MovieService.java
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public class MovieService implements IMovieService {

    private String serviceBaseUrl = "http://hickory-snap.hyperdev.space/";

    public JSONObject getAllMovies() throws IOException, JSONException {
        return JsonReader.readJsonFromUrl(serviceBaseUrl + "all");
    }
    public JSONObject getCurrentRentals(String username) throws IOException, JSONException {
        return JsonReader.readJsonFromUrl(serviceBaseUrl + "rentals?username="+ username);
    }
    public JSONObject rentMovie(String username, Integer movieId) throws IOException, JSONException {
        JSONObject payload = new JSONObject();
        payload.put("username", username);
        payload.put("id", movieId);
        return JsonPoster.postJsonToUrl(serviceBaseUrl + "rentMovie", payload.toString());
    }
    public JSONObject addMovie(String username, Movie movie) throws IOException, JSONException {
        JSONObject payload = new JSONObject();
        payload.put("username", username);
        payload.put("movie", movie);
        return JsonPoster.postJsonToUrl(serviceBaseUrl + "addMovie", payload.toString());
    }
}
